### DISCUSSION TITLE
send multiple email with pinpoint SendMessage

### DISCUSSION THREAD

**Discussion by juicycool92** [aws/aws-sdk-js-v3#4207] [Q&A] [answered]
Hello.

My goal is send multiple `email`s and `FCM push` with pinpoint.

method `sendMessage` is support multiple `fcm` and `email` address with `Addresses:Record<string, AddressConfiguration>`, thanks to me.
so i can manage to able to send multiple fcm messages. -not as exactly fit to my legacy, but ill handle it somehow-

next, i want to send multiple emails.
prepared `EmailTemplates`. I have several Templates in my case, depends of From user's type. 
I want to send difference Templates, to each user type.

my request bodies are like this..
```json
{
    "Addresses": {
        "FROM_USER@EMAIL.com": {
            "ChannelType": "EMAIL",
            "Context": {
                "uuid": "56bcc098-6197-43ed-80b1-4a48d3e833f0",
                "userType": "app"
            },
            "Substitutions": {
                "Trans.body.regards": [
                    "regards.."
                ],
                "Trans.subject": [
                    "[my_email_subject]test email title for APP!"
                ],
                "Trans.body.p1": [
                    "Hello world!"
                ],
                "Trans.body.p2": [
                    "this is body!"
                ]
            },
            "ReceiptHandle":" SOMETHING_RECEIPT_HANDLE"
        },
        "FROM_USER2@EMAIL.com": {
            "ChannelType": "EMAIL",
            "Context": {
                "uuid": "2a6ec24b-f002-4fe0-bf63-495fdeb1c440",
                "userType": "web"
            },
            "Substitutions": {
                "Trans.body.regards": [
                    "Regardsssss"
                ],
                "Trans.subject": [
                    "[my_email_subject]test email title for APP!"
                ],
                "Trans.body.p1": [
                    "Hello, <br>."
                ],
                "Trans.body.p2": [
                    "You made it!"
                ],
                "Trans.body.p3": [
                    "another text but dunno what to fill more"
                ]
            },
            "ReceiptHandle": "ANOTHER_RECEIPT"
        }
    },
    "Context": {},
    "Endpoints": {},
    "MessageConfiguration": {
        "EmailMessage": {
            "FromAddress": "admin@my_service.com"
        }
    },
    "TemplateConfiguration": {
        "EmailTemplate": {
            "Name": "MESSAGE_FOR_APP"
        }
    },
    "TraceId": "fffffff-ffff-ffff-ffffff"
}
```

as you can see the above example, there is two FROM users to send.
`FROM_USER@EMAIL.com` and `FROM_USER2@EMAIL.com`
each users has difference `Context.userType`.
and if userType is `app`, i want to set `TemplateConfiguration.EmailTemplate.Name as MESSAGE_FOR_APP`.
if userType is `web`, `TemplateConfiguration.EmailTemplate.Name as MESSAGE_FOR_WEB` .

but i cannot find any option about this so far.
is there something i missing? or this pinpoint doesn't support it and don't have plan for this?

Thanks.

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @juicycool92 - sorry for not getting here sooner. 

This's more of service related question rather than SDK but from my knowledge, your use case is unique and I don't think that feature is currently available. I would also recommend checking out SQS [sendMessage](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html) as an alternative.

If you're still working on finding the answer, I would be happy to assist you further with some guidance from service team.

Best,
John

**Reply by juicycool92** [reply]
Hello John,

As you mentioned before
Yes, i use SQS instead.

Thankyou for kind reply.
Have a nice day.
